vnet/udpproxy.go

```diff
@@ -82,6 +86,23 @@ type aUDPProxyWorker struct {
 	// key is vnet client addr, which is net.Addr
 	// value is *net.UDPConn
 	endpoints sync.Map
+
+	// For cleanup.
+	ctxDispose       context.Context
```

Comment:
https://golang.org/pkg/context/#pkg-overview
Storing Context inside struct is strongly discouraged in the official document.

Reply:
Yes, it's the best way to use context, the typical use scenario is pass `ctx context.Context` in the first param, like this:

```
func (v *UDPProxy) Proxy(ctx context.Context, client *Net, server *net.UDPAddr) error

func (v *UDPProxy) Close(ctx context.Context) error
```

But, it will change the interface of `io.Closer`, and the API of vnet which does not use `ctx context.Context` as the first param.  For example, `http.Server` changed the API and introduce a new func `Shutdown`:

```go
func (srv *Server) Close() error
func (srv *Server) Shutdown(ctx context.Context) error
```

There is some use case to keep the context as field of struct, for example:

```
type Request struct {
	ctx context.Context
}

func (r *Request) WithContext(ctx context.Context) *Request
``

So, I think right now, it's ok to use context in the struct, or as the first param of API.